lib/ansible/module_utils/network/iosxr/utils/utils.py

```diff
@@ -11,6 +11,56 @@
 from ansible.module_utils.six import iteritems
 
 
+def remove_command_from_interface(interface, cmd, commands):
+    # To delete the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append('no %s' % cmd)
+    return commands
+
+
+def add_command_to_interface(interface, cmd, commands):
+    # To set the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append(cmd)
+
+
+def dict_diff(sample_dict):
+    # Generate a set with passed dictionary for comparison
+    test_dict = {}
+    for k, v in iteritems(sample_dict):
+        if v is not None:
+            test_dict.update({k: v})
+    return_set = set(tuple(test_dict.items()))
+    return return_set
+
+
+def filter_dict_having_none_value(want, have):
+    # Generate dict with have dict value which is None in want dict
+    test_dict = dict()
+    test_dict['name'] = want.get('name')
+    for k, v in iteritems(want):
+        if v is None:
+            val = have.get(k)
+            test_dict.update({k: val})
+    return test_dict
+
+
+def remove_duplicate_interface(commands):
+    # Remove duplicate interface from commands
+    set_cmd = []
+    for each in commands:
+        if 'interface' in each:
+            interface = each
```

Comment:
What is the purpose of assigning to `interface` variable? `each` can be used directly in next line

Reply:
Also, this function can be made more generic by adding the substring as an argument. Something similar to 
```

def remove_duplicate_entries(commands, section):
    # Remove duplicate interface from commands
    set_cmd = []
    for each in commands:
        if section in each:
```

Reply:
I agree that implementing the fn. with section would make this generic, but I wrote this fn. for removing duplicate interface entries for replace and override state, so can I keep it as is with 1st comment implemented